src/main/java/com/github/krzysiekjodlowski/ox/Player.java

```diff
@@ -0,0 +1,61 @@
+package com.github.krzysiekjodlowski.ox;
+
+import com.github.krzysiekjodlowski.ox.model.Symbol;
+import com.github.krzysiekjodlowski.ox.ui.UI;
+
+/**
+ * Base for any player. Each one descendant
+ * will have to implement its own version
+ * of making moves and adjust toString.
+ *
+ * @author Krzysztof Jodlowski
+ */
+abstract class Player {
+  private final Symbol playersSymbol;
+
+  Player(Symbol symbol) {
+    this.playersSymbol = symbol;
+  }
+
+  Symbol getPlayersSymbol() {
+    return this.playersSymbol;
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (o == this) {
+      return true;
+    }
+    if (!(o instanceof Player)) {
+      return false;
+    }
+    Player that = (Player) o;
+    return this.playersSymbol == that.playersSymbol;
+  }
+
+  @Override
+  public int hashCode() {
+    return this.playersSymbol.hashCode();
+  }
+
+  /**
+   * Basic implementation. It is recommended for each
+   * child class to adjust this method to provide more
+   * specific information about its type.
+   *
+   * @return String representation of player
+   */
+  @Override
+  public String toString() {
+    return String.format("Player %s", this.playersSymbol);
+  }
+
+  /**
+   * Core functionality of player, which should
+   * be different in each type of a Player.
+   *
+   * @param ui generic user interface
+   * @return Move made by player
+   */
+  public abstract Move makeMove(UI<String, Integer> ui) throws NumberLowerThanOneException;
```

Comment:
UI in player class. It doesn't look nice for me. I think you should consider delegate such a method to a game class. In my opinion Player should be just a class to represent player's data (symbol, name, score - that kind of stuff) with handling basic operations like adding win or draw. When you make a move you need some board properties (maximum index of board) and player doesn't hold these properties and also doesn't hold a reference to the board in your implementation.

Reply:
You've got the point here, my bad. Moved collecting move logic to game class.